### PR TITLE
Config Apply Diagrams

### PR DESCRIPTION
**Pull request by aphralG**
### Proposed changes

Added flowchart and states for config apply
Stopped the Docs folder being deleted when running make clean 

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [ ] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works
- [ ] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [ ] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

File: docs/architecture/config_apply_state.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+flowchart TB
+ subgraph subGraph0["Legend"]
+        12["Internal Agent Status"]
+        13["Data Plane Response Status"]
+  end
+    0["COMMAND_STATUS_IN_PROGRESS"] --> 1["COMMAND_STATUS_IN_OK"]
+    2["COMMAND_STATUS_IN_PROGRESS"] --> 3["COMMAND_STATUS_IN_FAILURE"]
+    4["COMMAND_STATUS_IN_PROGRESS"] --> 5["COMMAND_STATUS_ERROR"]
+    5 --> 6["COMMAND_STATUS_IN_PROGRESS"]
+    6 --> 7["COMMAND_STATUS_ERROR"]
```
Comment (dhurley) [inline docs/architecture/config_apply_state.md:13]: should there be a `COMMAND_STATUS_FAILURE` after `COMMAND_STATUS_ERROR`?

File: docs/architecture/config_apply_state.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Config Apply States 
+```mermaid
+
+flowchart TB
+ subgraph subGraph0["Legend"]
+        12["Internal Agent Status"]
+        13["Data Plane Response Status"]
+  end
+    0["COMMAND_STATUS_IN_PROGRESS"] --> 1["COMMAND_STATUS_IN_OK"]
+    2["COMMAND_STATUS_IN_PROGRESS"] --> 3["COMMAND_STATUS_IN_FAILURE"]
```
Comment (dhurley) [inline docs/architecture/config_apply_state.md:10]:
```suggestion
    2["COMMAND_STATUS_IN_PROGRESS"] --> 3["COMMAND_STATUS_FAILURE"]
```

File: docs/architecture/config_apply_state.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# Config Apply States 
+```mermaid
+
+flowchart TB
+ subgraph subGraph0["Legend"]
+        12["Internal Agent Status"]
+        13["Data Plane Response Status"]
+  end
+    0["COMMAND_STATUS_IN_PROGRESS"] --> 1["COMMAND_STATUS_IN_OK"]
```
Comment (dhurley) [inline docs/architecture/config_apply_state.md:9]:
```suggestion
    0["COMMAND_STATUS_IN_PROGRESS"] --> 1["COMMAND_STATUS_OK"]
```

File: docs/architecture/config_apply.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    0["Start"] --> 2["Receive Config Apply Request"]
+    2 --> 3{"File in allowed directory list?"}
+    3 -- Yes --> 4["Compare File Hash"]
+    3 -- No --> 5["Send Data Plane Response <br> COMMAND_STATUS_FAILURE"]
+    5 --> 6["Clear File Cache"]
+    6 --> 1["End"]
+    4 --> 7{"Error Reading Files to Compare Hashes?"}
+    7 -- Yes --> 5
+    7 -- No --> 8["File Action Write, Add, Delete"]
+    8 --> 9{"Is Config Apply"}
```
Comment (dhurley) [inline docs/architecture/config_apply.md:27]: Not clear what `Is Config Apply` is referring to

**Review by oliveromahony — APPROVED** [review]

**Review by dhurley — APPROVED** [review]

**Pull request merged**